team/views.py

```diff
@@ -1,151 +1,20 @@
-
-        if member:
-            return Response(
-                {
-                    "status": "success",
-                    "results": serializer.data
-                },
-                status=status.HTTP_200_OK
-            )
-
-        return Response(
-            {
-                "status": "failed",
-                "results": ""
-            },
-            status=status.HTTP_404_NOT_FOUND
-        )
-
-    def post(self, request):
-        """
-        To add a team member
-        """
-        serializer = MemberSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(
-                {
-                    "status": "success",
-                    "results": "Successfully created a team member"
-                },
-                status=status.HTTP_201_CREATED
-            )
-
-        return Response(
-            {
-                "status": "failed",
-                "errors": serializer.errors,
-                "results": ""
-            },
-            status=status.HTTP_400_BAD_REQUEST
-        )
-
-    def delete(self, request):
-        kwargs = {}
-        userId = request.GET.get("user")
-        email = request.GET.get("email")
-
-        if userId is not None and userId != '':
-            kwargs["id"] = int(userId)
-
-        if email is not None and email != '':
-            kwargs["email"] = email
-
-        if len(kwargs):
-            member = self.get_object(kwargs)
-
-            if member:
-                member.delete()
-                return Response(
-                    {
-                        "status": "success",
-                        "results": "Successfully deleted a team member"
-                    },
-                    status=status.HTTP_200_OK
-                )
-
-        return Response(
-            {
-                "status": "failed",
-                "results": "Matching query not found"
-            },
-            status=status.HTTP_404_NOT_FOUND
-        )
-
-
-class EditMemberView(APIView):
-    def get_object(self, user_id):
-        try:
-            member = Member.objects.get(id=user_id)
-        except Member.DoesNotExist:
-            raise Http404
-        return member
-
-    def patch(self, request, user_id):
-        member = self.get_object(user_id)
-        serializer = MemberSerializer(member, data=request.data, partial=True)
-
-        if serializer.is_valid():
-            serializer.save()
-            return Response(
-                {
-                    "status": "success",
-                    "results": serializer.validated_data
-                },
-                status=status.HTTP_201_CREATED
-            )
-        return Response(
-            {
-                "status": "failed",
-                "errors": serializer.errors,
-                "results": ""
-            },
-            status=status.HTTP_201_CREATED
-        )
+    def patch(self, request, *args, **kwargs):
```

Comment:
I believe `RetrieveUpdateDestroyAPIView` uses the `UpdateModelMixin`, which supports patch: http://www.django-rest-framework.org/api-guide/generic-views/#updatemodelmixin

Reply:
Yes, it does. But the question says that I have to return only the updated properties. So that's why I'm overwriting the `patch` function.